modules/gdscript/language_server/gdscript_workspace.cpp

```diff
@@ -52,20 +52,13 @@ void GDScriptWorkspace::_bind_methods() {
 void GDScriptWorkspace::remove_cache_parser(const String &p_path) {
 	Map<String, ExtendGDScriptParser *>::Element *parser = parse_results.find(p_path);
 	Map<String, ExtendGDScriptParser *>::Element *script = scripts.find(p_path);
-	if (parser && script) {
-		if (script->get() && script->get() == script->get()) {
```

Comment:
Change to 
```
if (script->get() && script->get() == parser ->get()) {
```
should be ok.

Reply:
Done.
and thanks for the review

Reply:
@Geequlim See the previous discussion though, there's a lot of redundant code in 
https://github.com/godotengine/godot/blob/07c99340d1253799675cdf341c633c29ad0164ad/modules/gdscript/language_server/gdscript_workspace.cpp#L55-L70